### PR TITLE
check if correct adapter is used

### PR DESCRIPTION
**Pull request by tobiasKaminsky**

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
LGTM

File: src/third_parties/in/srain/cube/GridViewWithHeaderAndFooter.java [hunk 1]
```diff
@@ -412,6 +412,11 @@ public void setNumColumns(int numColumns) {
         }
     }
 
+    public boolean isCorrectAdapter(){
+        return getAdapter() == null ||
+               (getAdapter() != null && getAdapter() instanceof HeaderViewGridAdapter);
```
Comment (przybylski) [inline src/third_parties/in/srain/cube/GridViewWithHeaderAndFooter.java]: You don't need to do nullchecking: http://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.20.2

**Comment by przybylski** [comment]
But is checking just for an adapter type a root cause of this crash? Maybe something else is leading to it.

File: src/third_parties/in/srain/cube/GridViewWithHeaderAndFooter.java (hunk 1 above)
Comment (tobiasKaminsky) [inline src/third_parties/in/srain/cube/GridViewWithHeaderAndFooter.java]: Right, removed.

**Comment by tobiasKaminsky** [comment]
https://github.com/nextcloud/android/blob/d6ecf06f39bcf72e2fcf0d3c3d158f14e98259fc/src/third_parties/in/srain/cube/GridViewWithHeaderAndFooter.java#L182
here it is checked if mAdapter instanceof HeaderViewGridAdapter and otherwise it results in an exception.
The root cause is that we switch between ListAdapter and HeaderViewGridAdapter. Sometimes loading a directory takes too long and therefore it gets a race conflict and the old adapter is still in use, but the new "addFooterView" method is already called.

**Comment by przybylski** [comment]
👍 but i still thinks this isn't the way to fix it

**Comment by tobiasKaminsky** [comment]
We have some time till release, so feel free to fix it in a better way.
I also think that is not a solution, but a workaround...

**Comment by tobiasKaminsky** [comment]
@przybylski I suggest to put this in RC2, but try to get a better solution?

**Comment by AndyScherzinger** [comment]
Sounds good to me :+1:

**Comment by przybylski** [comment]
ok to me 👍

**Comment by AndyScherzinger** [comment]
review check seems to have a hickup...

**Comment by AndyScherzinger** [comment]
:+1:

**Comment by AndyScherzinger** [comment]
:+1:

**Comment by AndyScherzinger** [comment]
merged, will be shipped with 1.1.0 RC2

**Pull request merged**